packages/manager/modules/netapp/src/dashboard/volumes/dashboard/snapshots/add/controller.js

```diff
@@ -0,0 +1,36 @@
+export default class NetAppVolumesDashboardSnapshotsAddController {
+  /* @ngInject */
+  constructor($http, $translate) {
+    this.$http = $http;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+  }
+
+  addSnapshot() {
+    this.isLoading = true;
+    return this.$http
+      .post(
+        `/storage/netapp/${this.serviceName}/share/${this.volumeId}/snapshot`,
+        {
+          name: this.name,
+          description: this.description,
+        },
+      )
+      .then(() =>
+        this.goBack(
+          this.$translate.instant('netapp_volumes_snapshots_add_success'),
+        ),
+      )
+      .catch((error) =>
+        this.goBack(
+          this.$translate.instant('netapp_volumes_snapshots_add_error', {
+            message: error.data?.message,
+          }),
+          'error',
+        ),
+      );
```

Comment:
finally block not required? to set this.isLoading = false;

Reply:
When `this.isLoading = false` is set, it may show the modal with the primary button enabled for a few seconds, until the parent view loads (in case of reloading of parent view)

Reply:
We're always exiting modal so not mandatory but i'll add it for consistency